File: problems/041-progressive-cost-budget/solutions/python/main.py
import sys


values = list(map(int, sys.stdin.buffer.read().split()))
n, q = values[:2]
costs = values[2:2 + n]
budgets = values[2 + n:2 + n + q]

completed = 0
spent = 0
answers = []
for budget in budgets:
    while completed < n and costs[completed] <= budget - spent:
        spent += costs[completed]
        completed += 1
    answers.append(str(completed))

sys.stdout.write("\n".join(answers) + "\n")

File: problems/041-progressive-cost-budget/solutions/javascript/main.js
import * as std from "std";

const tokens = std.in.readAsString().trim().split(/\s+/);
let position = 0;
const n = Number(tokens[position++]);
const q = Number(tokens[position++]);
const costs = Array.from({ length: n }, () => BigInt(tokens[position++]));

let completed = 0;
let spent = 0n;
const answers = [];
for (let query = 0; query < q; query++) {
  const budget = BigInt(tokens[position++]);
  while (completed < n && costs[completed] <= budget - spent) {
    spent += costs[completed++];
  }
  answers.push(String(completed));
}

std.out.puts(answers.join("\n") + "\n");
